core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -39,12 +45,13 @@ public OutputStream compress(OutputStream os) throws IOException {
     }
   };
 
-  private static final ConcurrentMap<String, Decompressor> decompressors =
+  private static final ConcurrentMap<String, DecompressorInfo> decompressors =
       initializeDefaultDecompressors();
 
   /**
    * Represents a message compressor.
    */
+  @ExperimentalApi
```

Comment:
I honestly don't expect people to check the annotation anyway; I would imagine they would just use IDE autocomplete with reckless abandon.  Putting it here mostly a CYA, try to prevent as much support work for us.  

I agree it's too much; we should try to move it out of experimental!